src/app/shared/components/template/components/drawer/drawer.component.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Component,
+  ElementRef,
+  Output,
+  ViewChild,
+  EventEmitter,
+  AfterViewInit,
+} from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+
+@Component({
+  selector: "plh-drawer",
+  templateUrl: "./drawer.component.html",
+  styleUrls: ["./drawer.component.scss"],
+})
+export class TmplDrawerComponent extends TemplateBaseComponent implements AfterViewInit {
+  @ViewChild("drawer", { read: ElementRef }) drawer: ElementRef;
+
+  isOpen = false;
+  openHeight = 0;
+
+  async ngAfterViewInit() {
+    // hack to only get element heights after they've rendered
+    setTimeout(async () => {
+      const drawer = this.drawer.nativeElement;
+      const handle = drawer.children[0];
+      this.openHeight = drawer.offsetHeight - handle.offsetHeight - 10;
+      console.log(this.openHeight);
+    }, 100);
+  }
+
+  toggleDrawer() {
+    const drawer = this.drawer.nativeElement;
+
+    if (this.isOpen) {
+      drawer.style.transition = "transform .3s ease-out";
+      drawer.style.transform = "";
+      this.isOpen = false;
+    } else {
+      drawer.style.transition = "transform .3s ease-out";
+      drawer.style.transform = `translateY(${-this.openHeight}px)`;
+      this.isOpen = true;
+    }
+  }
+
+  // WIP: open and close drawer with a swipe
+  // Requires `gestureCtrl: GestureController` in constructor
+  // async enableGestures() {
```

Comment:
nit(non-blocking)
Have you tested this? What part of the code is still WIP?

The challenge with using a swipe gesture is there isn't really any visual change while swiping, you kind of just do an entire gesture and then the component disappears after. It might be better using either something like an [ion-item-sliding](https://ionicframework.com/docs/api/item-sliding) to make the card title slideable and auto-dismiss (like most mobile email clients), or use swiperjs  to create some sort of stacking effect
https://swiperjs.com/demos#effect-cards

Reply:
I think "swipe" is probably the wrong word – I was trying to enable the user to click and drag the top handle to open and close the drawer. This would also include a "flick" action that would send the drawer to its fully open/closed position without having to drag it all the way. Essentially the same behaviour as the bottom drawer in the Google Maps app, along the lines of [this React component](https://gorhom.github.io/react-native-bottom-sheet/).

It's proving quite complicated though, I've taken out the WIP code and can come back to it at a later date

Reply:
Ionic already has a component for that:
https://ionicframework.com/docs/api/modal#sheet-modal

If that doesn't work we could also consider:
https://panejs.com/